src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/CustomPipelineAction.java

```diff
@@ -0,0 +1,87 @@
+package com.northconcepts.datapipeline.foundations.examples.pipeline;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.northconcepts.datapipeline.core.DataReader;
+import com.northconcepts.datapipeline.core.Field;
+import com.northconcepts.datapipeline.core.FieldType;
+import com.northconcepts.datapipeline.core.Record;
+import com.northconcepts.datapipeline.foundations.file.LocalFile;
+import com.northconcepts.datapipeline.foundations.pipeline.Pipeline;
+import com.northconcepts.datapipeline.foundations.pipeline.action.PipelineAction;
+import com.northconcepts.datapipeline.foundations.pipeline.input.CsvPipelineInput;
+import com.northconcepts.datapipeline.foundations.pipeline.output.ExcelPipelineOutput;
+import com.northconcepts.datapipeline.transform.FieldTransformer;
+import com.northconcepts.datapipeline.transform.TransformingReader;
+
+public class CustomPipelineAction {
+
+    public static void main(String[] args) {
+        CsvPipelineInput pipelineInput = new CsvPipelineInput()
+                .setFileSource(new LocalFile().setPath("example/data/input/countries_with_country-code.csv"))
+                .setFieldNamesInFirstRow(true);
+ 
+        ExcelPipelineOutput pipelineOutput = new ExcelPipelineOutput()
+                .setFileSink(new LocalFile().setPath("example/data/output/countries_with_country-code.xlsx"))
+                .setFieldNamesInFirstRow(true);
+ 
+        Pipeline pipeline = new Pipeline()
+            .setInput(pipelineInput)
+            .setOutput(pipelineOutput)
+            .addAction(new UpcaseFieldsAction().addField("Country"));
+        
+        pipeline.run();
+    }
+    
+}
+
+class UpcaseFieldsAction extends PipelineAction {
```

Comment:
Please make this class a static inner class of CustomPipelineAction.

Reply:
Done